### DISCUSSION TITLE
how to pass Content Item Id to workflow through http post

### DISCUSSION THREAD

**Discussion by 13Groszy** [OrchardCMS/OrchardCore#16204] [Q&A] [answered]
Hi

I've got few content items which values need to be updated whenever Http post is made to the workflow. 
I've managed to update the values but only if Content Item ID passed to 'Update Content Task' is hardcoded. 

![Screenshot 2024-05-31 at 12 59 11](https://github.com/OrchardCMS/OrchardCore/assets/48838387/7b35b9c6-96a7-4e55-9114-7e1d5e87d869)

I'd like to pass Content Item ID dynamically with body/headers through Http post same as I pass value to change.
How can I archive that? Any tips? Unfortunately it has to be JavaScript expression so I can't pass direct values from Liquid.

Thanks

**Comment by douwinga** [comment, marked as answer]
You should be able to POST to the workflow endpoint with data in the body and then parse the body via `deserializeRequestData()`. What I typically do is use a "Set Property Task" to deserialize the request data by setting the Value to `deserializeRequestData();` . Then you have access to what was in the request body using something like `property('YourPropertyName').ContentItemId` in that "Content Item ID" field you show in your screenshot.
